src/uuid/uuid.nim

```diff
@@ -0,0 +1,67 @@
+import std/[strformat]
+import pkg/[uuids]
+import ".."/[logger]
+
+proc uuid*(n: Positive) =
+  ## Writes `n` version 4 UUIDs to stdout. Outputs only 1000 UUIDs if `n` is
+  ## greater than 1000.
+  const maxNumUuids = 1000
+  if n > maxNumUuids:
+    logNormal &"The UUID output limit is 1000, but {n} UUIDs were requested."
+    logNormal &"Outputting {maxNumUuids} UUIDs:"
+  for i in 1 .. min(n, maxNumUuids):
+    stdout.writeLine genUUID()
+
+func isValidUuidV4*(s: string): bool =
+  ## Returns `true` if `s` is a valid version 4 UUID (compliant with RFC 4122)
+  ## in the canonical textual representation.
+  ##
+  ## This func is equivalent to using `re.match` with the below regex pattern:
+  ##
+  ## `^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$`
+  ##
+  ## However, this func's implementation is faster and doesn't add a PCRE
+  ## dependency.
```

Comment:
The PCRE dependency is a good argument, the speed argument less so I feel as that won't be noticeable by the user most likely.

Reply:
Yes, the main argument is against PCRE and increasing the release binary size.

The speed difference is large, but far below what a user can perceive.

Reply:
Binary size is not a huge issue I think, but PCRE is essentially a dependency, right? As in: it has to be installed on the machine.

Reply:
Yes, but I meant that it's also possible to statically link PCRE. I haven't checked exactly how much that'd increase the binary size, but it'd be significant.

At the moment we use `--passL:static` on Linux and Windows:

https://github.com/exercism/canonical-data-syncer/blob/11ab015b4d6dcb692e353028b80882af60d91f3a/.github/bin/build#L6-L13

Reply:
Okay cool